instrumentation/jms/src/main/java/brave/jms/PropertyFilter.java

```diff
@@ -40,31 +42,39 @@ static void filterProperties(Message message, Set<String> namesToClear) {
   }
 
   static void filterProperties(Message message, Set<String> namesToClear, List<Object> out) {
-    Enumeration<?> names;
+    List<String> names;
     try {
-      names = message.getPropertyNames();
+      names = list(message.getPropertyNames());
     } catch (Throwable t) {
       propagateIfFatal(t);
       log(t, "error getting property names from {0}", message, null);
       return;
     }
-
-    while (names.hasMoreElements()) {
-      String name = (String) names.nextElement();
-      Object value;
-      try {
-        value = message.getObjectProperty(name);
-      } catch (Throwable t) {
-        propagateIfFatal(t);
-        log(t, "error getting property {0} from message {1}", name, message);
-        return;
-      }
-      if (!namesToClear.contains(name) && value != null) {
-        out.add(name);
-        out.add(value);
+    
+    boolean disjoint = true;
+    for (String name: names) {
+      if (!namesToClear.contains(name)) {
```

Comment:
small nit: use early termination for easier readability: 
```java
for (String name: names) {
      if (namesToClear.contains(name)) {
          disjoint = false;
          continue;
      }
// all the rest

Reply:
I usually use it in methods, but I wanted to avoid "continue".

Reply:
refactored as you proposed